tests/Adapters/Microsoft.Bot.Builder.Adapters.Twilio.Tests/TwilioAdapterTests.cs

```diff
@@ -16,57 +16,28 @@ namespace Microsoft.Bot.Builder.Adapters.Twilio.Tests
     public class TwilioAdapterTests
     {
         [Fact]
-        public void Constructor_Should_Fail_With_Null_Options()
-        {
-            Assert.Throws<ArgumentNullException>(() => { new TwilioAdapter(null, new Mock<TwilioClientWrapper>().Object); });
-        }
-
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_TwilioNumber()
-        {
-            var options = new TwilioAdapterOptions(null, "Test", "Test", new Uri("http://contoso.com"));
-
-            Assert.Throws<ArgumentException>(() => { new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object); });
-        }
-
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_AccountSid()
-        {
-            var options = new TwilioAdapterOptions("Test", null, "Test", new Uri("http://contoso.com"));
-
-            Assert.Throws<ArgumentException>(() => { new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object); });
-        }
-
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_AuthToken()
-        {
-            var options = new TwilioAdapterOptions("Test", "Test", null, new Uri("http://contoso.com"));
-
-            Assert.Throws<ArgumentException>(() => { new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object); });
-        }
-
-        [Fact]
-        public void Constructor_Should_Fail_With_Null_TwilioApi()
+        public void Constructor_Should_Fail_With_Null_TwilioAPIWrapper()
         {
             var options = new TwilioAdapterOptions("Test", "Test", "Test", new Uri("http://contoso.com"));
+            TwilioClientWrapper twilioClientWrapper = null;
 
-            Assert.Throws<ArgumentNullException>(() => { new TwilioAdapter(options, null); });
+            Assert.Throws<ArgumentNullException>(() => { new TwilioAdapter(twilioClientWrapper); });
         }
 
         [Fact]
-        public void Constructor_WithArguments_Succeeds()
+        public void Constructor_With_TwilioAPIWrapper_Succeeds()
         {
             var options = new TwilioAdapterOptions("Test", "Test", "Test", new Uri("http://contoso.com"));
 
-            Assert.NotNull(new TwilioAdapter(options, new Mock<TwilioClientWrapper>().Object));
+            Assert.NotNull(new TwilioAdapter(new Mock<TwilioClientWrapper>(options).Object));
```

Comment:
We should have the options as a property to use them in all tests without duplicating code.
```suggestion
            Assert.NotNull(new TwilioAdapter(new Mock<TwilioClientWrapper>(_testOptions).Object));
```